Purpose:
Explore customer behaviour and identify patterns associated with churn to inform retention strategies.

In [1]:
import pandas as pd

df = pd.read_csv("../data/clean_telco_churn.csv")
df.head()

customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0  7590-VHVEG  Female              0     Yes         No       1           No   
1  5575-GNVDE    Male              0      No         No      34          Yes   
2  3668-QPYBK    Male              0      No         No       2          Yes   
3  7795-CFOCW    Male              0      No         No      45           No   
4  9237-HQITU  Female              0      No         No       2          Yes   

      MultipleLines InternetService OnlineSecurity  ... DeviceProtection  \
0  No phone service             DSL             No  ...               No   
1                No             DSL            Yes  ...              Yes   
2                No             DSL            Yes  ...               No   
3  No phone service             DSL            Yes  ...              Yes   
4                No     Fiber optic             No  ...               No   

  TechSupport StreamingTV StreamingMovies        Contract PaperlessBilling  \
0          No          No              No  Month-to-month              Yes   
1          No          No              No        One year               No   
2          No          No              No  Month-to-month              Yes   
3         Yes          No              No        One year               No   
4          No          No              No  Month-to-month              Yes   

               PaymentMethod MonthlyCharges  TotalCharges  Churn  
0           Electronic check          29.85         29.85      0  
1               Mailed check          56.95       1889.50      0  
2               Mailed check          53.85        108.15      1  
3  Bank transfer (automatic)          42.30       1840.75      0  
4           Electronic check          70.70        151.65      1  

[5 rows x 21 columns]

In [2]:
df['Churn'].mean()

np.float64(0.26578498293515357)

Overall Churn Rate
Approximately 26.6% of customers have churned, indicating a meaningful retention issue that warrants further investigation into customer behaviour and service characteristics.

🧩 Question 1 — Does contract type affect churn?

In [3]:
df.groupby('Contract')['Churn'].mean().sort_values(ascending=False)


Contract
Month-to-month    0.427097
One year          0.112772
Two year          0.028487
Name: Churn, dtype: float64

Churn by Contract Type
Customers on month-to-month contracts churn at a significantly higher rate (~43%) compared to one-year (~11%) and two-year (~3%) contracts. This suggests that contractual commitment is a major driver of customer retention.

🔍 Question 2 — Does tenure influence churn?

In [4]:
df['tenure_group'] = pd.cut(
    df['tenure'],
    bins=[0, 6, 12, 24, 60],
    labels=['0-6 months', '6-12 months', '12-24 months', '24+ months']
)

In [5]:
df.groupby('tenure_group')['Churn'].mean()


tenure_group
0-6 months      0.533333
6-12 months     0.358865
12-24 months    0.287109
24+ months      0.183430
Name: Churn, dtype: float64

Churn by Tenure
Customer churn is highest during the early stages of the customer lifecycle, with over 53% of customers in their first 6 months churning. Churn steadily decreases as tenure increases, suggesting that improving early onboarding and initial engagement could significantly reduce overall churn.

🔍 Question 3 — Does pricing intensity affect churn?

In [6]:
df['charge_group'] = pd.qcut(
    df['MonthlyCharges'],
    q=4,
    labels=['Low', 'Mid-Low', 'Mid-High', 'High']
)

In [7]:
df.groupby('charge_group')['Churn'].mean()

charge_group
Low         0.113197
Mid-Low     0.245883
Mid-High    0.375499
High        0.328783
Name: Churn, dtype: float64

Churn by Monthly Charges
Customers with higher monthly charges exhibit higher churn rates, particularly in the mid-high pricing segment. This suggests price sensitivity plays a role in churn, especially when perceived value does not scale with cost.

📌 Key Findings                                                                   
Overall churn rate is approximately 26.6%, indicating a significant retention challenge.

Month-to-month contracts have a substantially higher churn rate (~43%) compared to longer-term contracts.

Customer churn is highest within the first 6 months of tenure and decreases steadily with longer retention.

Higher monthly charges are associated with increased churn, suggesting potential price sensitivity among higher-paying customers.                  